### ISSUE TITLE
[wasm64] wrong type for st_nlink in doStat

### ISSUE THREAD

**Issue by tiran** [emscripten-core/emscripten#17620]
**Version of emscripten/emsdk:**
```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.18 (d5ca9bba6513763d5bdddbd0efff759332bd85d7)
clang version 16.0.0 (https://github.com/llvm/llvm-project 48129cf0ed5053ed3fdf4f928180635e84892614)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /opt/emsdk/upstream/bin
```

**issue:**
```
#include <sys/stat.h>
#include <stdio.h>

int main(void) {
#ifdef __wasm64__
    puts("wasm64");
#endif
    struct stat st;
    printf("st_nlink: %lu, %lu\n", sizeof(st.st_nlink), offsetof(struct stat, st_nlink));
    return 0;
}
```

```
$ emcc -sMEMORY64=1 struct.c -o struct.js; /usr/bin/node --experimental-wasm-memory64 struct.js
wasm64
st_nlink: 8, 24
```

The type conversion for ``struct stat.st_nlink`` is incorrect. ``doStat`` converts the field member as ``i32``. The ``nlink_t``is 64bit type on WASM64.

https://github.com/emscripten-core/emscripten/blob/2b2e142900c4512a4e01972f4a3a2c2c025da3a1/src/library_syscall.js#L55-L71

**Comment by tiran** [comment]
The code for ``st_[amc]tim.tv_nsec`` is also incorrect. It's a 64bit type on WASM64.

**Comment by sbc100** [comment]
Thanks for the report!  if you would like to upload a PR it would be most welcome.  

As you can see wasm64 is will a work in progress so there are likely many issues like this.   I'm working getting more and more of the test suite to pass on wasm64 but there is still fair amount of work to do.

**Comment by tiran** [comment]
I have included a fix in PR #17613.

With fixes from the PR and another PR for CPython I'm able to run the Python test suite on WASM64 with just two test failures.

**Comment by kripken** [comment]
@tiran 

> With fixes from the PR and another PR for CPython I'm able to run the Python test suite on WASM64 with just two test failures.

Great, that's very exciting!
